### ISSUE TITLE
Deep learning 코드 복습할 때 보는 목차

### ISSUE THREAD

**Issue by leggiero-crescendo** [leggiero-crescendo/Dev-perception#6]
# Image classification 순서
1. Get dataset
- torch vision 지원되는 걸로 사용 했지만,,,!! 아닌건 입력되는 방법 찾아두기(뒤에 나올 듯)
- transform을 해야합니다.(raw data를 변환하여 가져오는)

2. DataLoader
- 학습 데이터 준비

3. Model 설계
- Lenet5 kernel과 layer 정보 !  forward 연산선언 !!

4. main에서 train 설정
- model 불러오기 -> device 올리기
- optimizer & scheduler define
- loss function 설정해주기 (따로 파일에서 설정)
- epoch , iter define
- 하나의 epoch 안에서 -> 한번에 train_loader 에서 batch의 갯수에 따라 이미지와 인퍼런스 !!
- output 결과와 gt결과에 따라 loss 계싼 
- backpropagation 진행(loss_val.backward()->optimizer.step -> zero_grad -> total loss 계산)
- 한 epoch 마다 scheduler step , torch.save(model.state_dict(), args.output_dir,~~pt) 
5. eval
- model 부르기
- save 형태에 따라 load 방식이 달라지겠지만 위의 형태일때는 
```python
checkpoint = torch.load(args.checkpoint)
model.load_state_dict(checkpoint)
model.to(device)
model.eval()
```
- model inference 만 해줌 out 과 gt 비교함
